src/ui/digital-exchange/components/SearchBar.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Icon } from 'patternfly-react';
+import { throttle } from '@entando/utils';
+
+class SearchBar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: '',
+      filled: false,
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.handleClear = this.handleClear.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({
+      value: e.target.value,
+      filled: (e.target.value !== ''),
+    });
+  }
+
+  handleClear() {
+    const triggerSubmit = this.state.value !== '';
+    this.setState({
+      value: '',
+      filled: false,
+    });
+    if (triggerSubmit) {
+      throttle(() => this.handleSubmit());
+    }
+  }
+
+  handleKeyUp(e) {
+    if (e.keyCode === 13) this.handleSubmit();
```

Comment:
`keyCode` is deprecated https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

use `key` instead

Reply:
done @gidesan